### DISCUSSION TITLE
Enabling User Registration in Superset

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#13266] [Q&A / Help] [answered]
I am trying to use User Registration feature.
When user tries to register on Superset, the UI shows error:
```
Not possible to register you at the moment, try again later
Access is Denied
```

#### Screenshots

![image](https://user-images.githubusercontent.com/74634977/108645446-35e66300-74f6-11eb-82e8-ba15727d5264.png)

#### How to reproduce the issue

1. Enabled User Registration from config `AUTH_USER_REGISTRATION=True` with authentication type as `AUTH_DB` and `AUTH_USER_REGISTRATION_ROLE = "admin"`
2. From register screen, fill all details and click on 'Submit'
3. The user is send back to login screen with error as seen in screenshot below.

Superset logs show only 1 line error:
```
ERROR:flask_appbuilder.security.registerviews:Send email exception: [Errno 61] Connection refused
```

### Environment

- superset version: `v1.0.1`
- python version: `3.7`
- node.js version: `v14.15.5`

### Additional context

It seems like Superset is trying to send email to newly registering user, but not sure from which email. I have setup email for Email notifications and that feature is working fine.
Is there any other config missing?

**Comment by kamalkeshavani-aiinside** [comment, marked as answer]
Finally from Flask documentation [here](https://flask-appbuilder.readthedocs.io/en/latest/user_registration.html#database-authentication), I found that I need this configuration in config.py

```
MAIL_SERVER = 'smtp.gmail.com'
MAIL_USE_TLS = True
MAIL_USERNAME = 'yourappemail@gmail.com'
MAIL_PASSWORD = 'passwordformail'
MAIL_DEFAULT_SENDER = 'fabtest10@gmail.com'
```

**Reply by atpage** [reply]
Also if you're using Gmail with TLS like that example, you'll need to set `MAIL_PORT = 587`.

**Reply by akzarma** [reply]
Also, don't forget to install Flask-Mail python lib.
